elastic/kibana
# [telemetry] update logic to fix misisng kibana stats when monitoring is just enabled

Monitoring Telemetry sends usage data with missing kibana usage stats for a maximum of a day if kibana starts with monitoring disabled and then it gets enabled.

This issue started happening when we modified the fetching logic of usage collectors to once per day instead of every 10 seconds. What happens is that we set the filterUsage boolean to true which tells monitoring to not upload usage stats to ES with the assumption that we have an older usage document contained in the index.

In the case monitoring is not enabled, we stop uploading usage stats for a day, any telemetry reported during that day after monitoring gets enabled will try to report kibana stats but the monitoring index does not contain any since we assumed the insert was initially successful (but it was ignored since there is no monitoring indices).

This PR fixes the bug by checking if the ES upload was failed/ignored then we keep trying to upload the usage stats.

The good news is that we are losing the kibana usage stats for a maximum of 24 hours after monitoring is enabled (depends on when the server was started). This logic also gets reset on every kibana restart.

In math terms it is something like this: (monitoring enabled time - server start time) % 24 hours

Closes https://github.com/elastic/telemetry/issues/126



## Testing

Reproducing this issue prior to this Fix:

1. Start a fresh kibana and ES instance
2. Start monitoring
3. Check telemetry example usage.
4. "kibana" usage property will be missing from the usage stats for the first 24 hours.

Restarting kibana will resolve this, or waiting 24 hours.
